src/pydata_sphinx_theme/toctree.py

```diff
@@ -35,9 +35,12 @@ def add_inline_math(node: Node) -> str:
 def get_unrendered_local_toctree(
     app: Sphinx, pagename: str, startdepth: int, collapse: bool = True, **kwargs
 ):
-    """."""
-    if "includehidden" not in kwargs:
-        kwargs["includehidden"] = False
+    """Get the "local" (starting at `startdepth`) TOC tree containing the current page.
+
+    This is similar to `context["toctree"](**kwargs)` in sphinx templating,
+    but using the startdepth-local instead of global TOC tree.
+    """
+    kwargs.setdefault("includehidden", True)
```

Comment:
this is the key change. it was copy/pasted from its old place within `index_toctree` in #1609 and I don't really know why it was previously setting `False` as the default.  I'm not even sure that we *should* be forcing it here; it might be better to leave it unset (or better yet: to use the theme setting `sidebar_includehidden`)

Reply:
My feeling is that `includehidden` is sensible to default to True, so I'm +1 on this. A common use-case is to use `hidden` to not show the TOCtree on the page but people do want it in the sidebar.

Reply:
Yes but see #1551

Reply:
That just means we need it to be possible to not include them, but doesn't change your opinion that the default should be to show them, right?

Reply:
Agreed the default behviour is to include them in all the other theme I know so Users (and us) would be surprised not to see the complete toctree in the sidebar (again by default)

Reply:
> That just means we need it to be possible to not include them, but doesn't change your opinion that the default should be to show them, right?

Yes. But that means we need to use `theme_sidebar_includehidden` *somewhere* where is currently not being used, so that it ends up in these kwargs. I think it will work to do that in `layout.html` as in the linked PR but I haven't had time to do thorough testing/debugging yet.

But it's helpful to know that this fixes things for @lwasser --- it means we're on the right track